core/controller/src/main/scala/org/apache/openwhisk/core/controller/Controller.scala

```diff
@@ -175,6 +193,30 @@ class Controller(val instance: ControllerInstanceId,
     LogLimit.config,
     runtimes,
     List(apiV1.basepath()))
+
+  private val controllerUsername = loadConfigOrThrow[String](ConfigKeys.whiskControllerUsername)
+  private val controllerPassword = loadConfigOrThrow[String](ConfigKeys.whiskControllerPassword)
+
+  /**
+   * disable controller
+   */
+  private def disable(implicit transid: TransactionId) = {
+    implicit val executionContext = actorSystem.dispatcher
+    implicit val jsonPrettyResponsePrinter = PrettyPrinter
+    (path("disable") & post) {
+      extractCredentials {
+        case Some(BasicHttpCredentials(username, password)) =>
+          if (username == controllerUsername && password == controllerPassword) {
+            loadBalancer.close
```

Comment:
This looks like it shuts down the messaging feed to receive completionAcks from the invoker, but how does it stop the controller from receiving new activations which will then be orphaned and never be able to respond when it receives the completionAck after restarting.

I think it would be a good idea to have an `isEnabled` route similar to the other services so that the external monitor that is determining whether the service is alive to direct traffic to the node can make the decision to stop directing traffic once disabled.

Reply:
> I think it would be a good idea to have an isEnabled route similar to the other services so that the external monitor that is determining whether the service is alive to direct traffic to the node can make the decision to stop directing traffic once disabled.

That sounds good but not easy to add it with Nginx.
Take a look at [this issue](https://github.com/apache/openwhisk/issues/5337).